src/btree/bt_cursor.c

```diff
@@ -512,6 +512,39 @@ __wt_btcur_search_prepared(WT_CURSOR *cursor, WT_UPDATE **updp)
     return (0);
 }
 
+/*
+ * __wt_btcur_evict_reposition --
+ *     Try to evict the page and reposition the cursor on the saved key.
+ */
+int
+__wt_btcur_evict_reposition(WT_CURSOR_BTREE *cbt)
+{
+    WT_CURSOR *cursor;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+
+    cursor = &cbt->iface;
+    session = CUR2S(cbt);
+    F_CLR(cbt, WT_CBT_EVICT_REPOSITION);
+
+    /*
+     * Try to evict the page and then reposition the cursor back to the page for operations with
+     * snapshot isolation and for pages that require urgent eviction. Snapshot isolation level
+     * maintains a snapshot allowing the cursor to point at the correct value after a reposition
+     * unlike read committed isolation level.
+     */
+    if (session->txn->isolation == WT_ISO_SNAPSHOT && F_ISSET(session->txn, WT_TXN_RUNNING) &&
+      __wt_page_evict_soon_check(session, cbt->ref, NULL)) {
+        WT_RET(__wt_cursor_localkey(cursor));
+        WT_RET(__cursor_reset(cbt));
+        ret = __wt_btcur_search(cbt);
+        if (ret != 0 && ret != WT_ROLLBACK)
```

Comment:
Add a comment here describe the rollback case?

Reply:
Updated in [cf9ec97](https://github.com/wiredtiger/wiredtiger/pull/7687/commits/cf9ec9718b35e81c9ef9a41fd379272771f78c4a)